src/main/java/edu/ie3/datamodel/io/source/csv/CsvThermalGridSource.java

```diff
@@ -70,12 +70,16 @@ public static List<ThermalGrid> read(
     Try<Collection<ThermalHouseInput>, SourceException> houses =
         Try.of(
             () -> thermalSource.getThermalHouses(operators, buses).values(), SourceException.class);
-    Try<Collection<ThermalStorageInput>, SourceException> storages =
+    Try<Collection<ThermalStorageInput>, SourceException> heatStorages =
+        Try.of(
+            () -> thermalSource.getThermalStorages(operators, buses).values(),
+            SourceException.class);
+    Try<Collection<ThermalStorageInput>, SourceException> waterStorages =
         Try.of(
             () -> thermalSource.getThermalStorages(operators, buses).values(),
             SourceException.class);
```

Comment:
Both collections will contain all storages (cylindrical and hot water). I think you need to call the specific methods.